server/src/services/metadata.service.ts

```diff
@@ -438,11 +444,13 @@ export class MetadataService {
         this.storageCore.ensureFolders(motionPath);
         await this.storageRepository.writeFile(motionAsset.originalPath, video);
         await this.jobRepository.queue({ name: JobName.METADATA_EXTRACTION, data: { id: motionAsset.id } });
-        await this.assetRepository.update({ id: asset.id, livePhotoVideoId: motionAsset.id });
+      }
 
+      if (asset.livePhotoVideoId !== motionAsset.id) {
+        await this.assetRepository.update({ id: asset.id, livePhotoVideoId: motionAsset.id });
```

Comment:
This should probably be done after deleting the old live photo asset `asset.livePhotoVideoId`

Reply:
The original code was in this order (update `livePhotoVideoId` before deleting old motion video assets)
I think this works because the `livePhotoVideoId` update is not reflected in `asset` immediately.

open to discussion

Reply:
Mhh yeah is see the original has this order aswell. Perhaps I am missing something but I think its wrong. What if the update to `livePhotoVideoId` happens before reaching the deletion part? It could delete the new `livePhotoVideoId` that was just updated.

Reply:
I doubt that's possible.
We don't re-fetch or update the existing `asset` object from the database